### ISSUE TITLE
unparam doesn't apply to function scope

### ISSUE THREAD

**Issue by AshleyRoll** [douglascrockford/JSLint#128]
I have reduced a problem that I am seeing with the unparam option to the following test cases.

I want to ignore the unused parameter "data" in "ignoreUnused" because this function will be attached as a handler from some third party library, but I want to have the unused parameter in "failUnused" reported.

I expected that I could simply put a /*jslint ... */ directive in the function declaration, but it does not work.

``` javascript
var a = (function () {
    'use strict';

    return {
        ignoreUnused: function (data, event) {
            /*jslint unparam:true */
            event.foo();
        },

        failUnused: function (data) {
        }
    };
}());
```

> > > Unused variable: data 5 'ignoreUnused', data 10 'failUnused'

The following ignores the unused parameter, but masks the error in failUnused.

``` javascript
var a = (function () {
    'use strict';
    /*jslint unparam:true */

    return {
        ignoreUnused: function (data, event) {
            event.foo();
        },

        failUnused: function (data) {
        }
    };
}());
```

It appears that the /*jslint ... */ directive is not being applied in this case to the function's scope, however for other options (like regexp) it does.
